v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-component.tsx

```diff
@@ -239,9 +240,14 @@ export const UnivariateMeasureAdornmentComponent = observer(
       const lineOffset = 5
       const topOffset = 50
       let x = isVertical.current ? xScale(plotValue) / xCellCount + lineOffset : (plotWidth - plotWidth/2) / xCellCount
-      if (range && isVertical.current) x = x + range
+      if (range && isVertical.current) x = xScale(plotValue + range) / xCellCount + lineOffset
       let y = isVertical.current ? topOffset : yScale(plotValue) / yCellCount - lineOffset
-      if (range && !isVertical.current) y = y - range
+      if (range && !isVertical.current) y = yScale(plotValue + range) / yCellCount - lineOffset
+
+      // If x plus the approximate width of the text tip would extend beyond the right boundary of the subplot, set x to
+      // plotWidth minus the text tip width or zero, whichever is greater.
+      const textTipWidth = textContent.length * 8
```

Comment:
This method for determining the width of the tool tip was suggested by Copilot. It won't be 100% accurate, but it seems reasonable to me as an approximation that will work most of the time. The only alternative I'm aware of would depend on adding a hidden element to the DOM, inserting the text into that element, and then checking that element's width. There's very possibly some other option, but it would probably depend on revamping how the text tips are constructed. If this seems reasonable enough, I'd rather not investigate that further right now.

Reply:
Did you look into the available `measureText` or `useMeasureText` utilities already in use?